contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checker.py

```diff
@@ -167,6 +172,19 @@ def checkstyle(self, sources):
 
   def execute(self):
     """Run Checkstyle on all found non-synthetic source files."""
+
+    # If we are linting for Python 3, skip lint altogether. 
+    # Long-term Python 3 linting solution tracked by:
+    # https://github.com/pantsbuild/pants/issues/5764
+    interpreter = self.context.products.get_data(PythonInterpreter)
+    # Check interpreter is not 'None' for test cases that do not
+    # run the python interpreter selection task.
+    if interpreter and interpreter.version > (3, 0 ,0):
```

Comment:
Although this will get everything we care about, `>= (3, 0, 0)` reads as more obviously correct.